routing/router.go

```diff
@@ -2155,6 +2156,127 @@ func (r *ChannelRouter) FindRoute(req *RouteRequest) (*route.Route, float64,
 	return route, probability, nil
 }
 
+// probabilitySource defines the signature of a function that can be used to
+// query the success probability of sending a given amount between the two
+// given vertices.
+type probabilitySource func(route.Vertex, route.Vertex, lnwire.MilliSatoshi,
+	btcutil.Amount) float64
+
+// BlindedPathRestrictions are a set of constraints to adhere to when
+// choosing a set of blinded paths to this node.
+type BlindedPathRestrictions struct {
+	// MinNumHops is the minimum number of hops to include in a blinded
+	// path. This doesn't include our node, so if the minimum is 1, then
+	// the path will contain at minimum our node along with an introduction
+	// node hop.
+	MinNumHops uint8
+
+	// MaxNumHops is the maximum number of hops to include in a blinded
+	// path. This doesn't include our node, so if the maximum is 1, then
+	// the path will contain our node along with an introduction node hop.
+	MaxNumHops uint8
+
+	// MaxNumPaths is the maximum number of blinded paths to select.
+	MaxNumPaths uint8
+}
+
+// FindBlindedPaths finds a selection of paths to the destination node that can
+// be used in blinded payment paths.
+func (r *ChannelRouter) FindBlindedPaths(destination route.Vertex,
```

Comment:
some food for thought here:

- If we let the user set a non-equal MinHops and MaxHops, then this gonna pretty much select the shortest paths as is.. since those will always have the largest success probability. So perhaps we should also have a MinNumPaths and then always only select paths of the same length as long there are enough with that length to satisfy the MinNumPaths?

Reply:
- we also in general probably dont want to include paths of diff lengths. Currently the user has the power to enforce this through just setting MaxNumHops = MinNumHops but I dont think they should have to do that

Reply:
wanna hear other people's thoughts before I update anything though

Reply:
update: after thinking about it for a bit, i think it might actually not be that complicated to add! gonna work on this today to see if I can make it happen 😈

Reply:
added dummy hops!

Reply:
> We would degrade the longer path's privacy to the shorter ones if one can assume that all intro nodes may be present in the other paths as relay nodes. A downside of that is that it restricts the number of possible paths, so it may not be possible to find some path. This also circumvents the problem with the apriori probability (shortest paths are selected first).

Ahh interesting point 👀, I would for the first release maybe focus on successful paths and introduce more privacy focused settings down the road ?